### DISCUSSION TITLE
How to change colors for `mode` and `diagnostic_count`?

### DISCUSSION THREAD

**Discussion by divramod** [MunifTanjim/nougat.nvim#74] [Q&A] [answered]
hey hey,

awesome plugin, thx for writing it!

is it possible to change the colors for the different modes like normal, visual, insert?
i looked at the 3 examples and wasn't able to find it?

the same goes for the diagnostic_count?
can i somewhere change these too, like i'll do it for the git counts?

thx again for the plugin!

**Comment by towry** [comment]
https://github.com/MunifTanjim/nougat.nvim/pull/6 isn't this what you want?

**Comment by MunifTanjim** [comment, marked as answer]
Yes absolutely.

---

For `mode`:

```lua
local color = require("nougat.color").get()

local mode = nut.mode.create({
  config = {
    highlight = {
      insert = { bg = color.blue, fg = color.bg },
      -- ...
    },
  },
})
```

Ref:
- https://github.com/MunifTanjim/nougat.nvim/blob/6c7f5428226ed1f8e4f40595c3babacecb77d436/lua/nougat/nut/mode.lua#L7-L16
- https://github.com/MunifTanjim/nougat.nvim/blob/6c7f5428226ed1f8e4f40595c3babacecb77d436/lua/nougat/nut/mode.lua#L67-L143

---

For `diagnostic_count`:

```lua
local diagnostic_count = nut.buf.diagnostic_count.create({
  hidden = nut.buf.diagnostic_count.hidden.if_zero(),
  hl = { bg = color.bg3 },
  prefix = " ",
  suffix = " ",
  config = {
    error = { prefix = " ", fg = color.red },
    warn = { prefix = " ", fg = color.yellow },
    info = { prefix = " ", fg = color.blue },
    hint = { prefix = "󰌶 ", fg = color.green },
  },
})
```

Ref:
- https://github.com/MunifTanjim/nougat.nvim/blob/6c7f5428226ed1f8e4f40595c3babacecb77d436/lua/nougat/nut/buf/diagnostic_count.lua#L10-L32

**Comment by MunifTanjim** [comment]
Here's an example for `git.status`:

https://github.com/MunifTanjim/nougat.nvim/blob/6c7f5428226ed1f8e4f40595c3babacecb77d436/examples/bubbly.lua#L148-L176
